core/src/main/java/org/apache/brooklyn/core/mgmt/BrooklynTags.java

```diff
@@ -59,6 +59,21 @@ public String getKind() {
         public String getContents() {
             return contents;
         }
+
+        @Override
+        public boolean equals(Object other) {
+            if (other == null || !(other instanceof NamedStringTag)) {
+                return false;
+            }
+
+            return kind != null && kind.equals(((NamedStringTag)other).kind)
+                    && contents != null && contents.equals(((NamedStringTag)other).contents);
+        }
+
+        @Override
+        public int hashCode() {
+            return (kind == null ? 0 : kind.hashCode()) + (contents == null ? 0 : contents.hashCode());
```

Comment:
Personal preference is to use guava's utilities: `Objects.hashCode(kind, contents)`